lib/moped/errors.rb

```diff
@@ -122,11 +122,12 @@ class PotentialReconfiguration < MongoError
       # not a master. (This encapsulates codes 10054, 10056, 10058)
       def reconfiguring_replica_set?
         err = details["err"] || details["errmsg"] || details["$err"] || ""
-        NOT_MASTER.include?(details["code"]) || err.include?("not master")
+        NOT_MASTER.include?(details["code"]) || err.include?("not master") || err.include?("could not contact primary")
       end
 
       def connection_failure?
-        CONNECTION_ERRORS_RECONFIGURATION.include?(details["code"])
+        err = details["err"] || details["errmsg"] || details["$err"] || ""
+        CONNECTION_ERRORS_RECONFIGURATION.include?(details["code"]) || err.include?("could not get last error") || err.include?("connection attempt failed") || err.include?("possible socket exception")
```

Comment:
erm, I mean in the first case where it could not force a write concern, no code was returned. I can add errors 7 and 83 as codes.